Expensify/expensify-common
# Enable display of empty blockquotes

First part of the blockquotes display-related prs. 

<img width="379" alt="image" src="https://github.com/Expensify/expensify-common/assets/45288762/0ab3996a-9e5c-4359-b690-6261555d7ddb">

<img width="416" alt="image" src="https://github.com/Expensify/expensify-common/assets/45288762/fa47da87-c748-4731-8a66-ad6000504c5f">

<img width="416" alt="image" src="https://github.com/Expensify/expensify-common/assets/45288762/3a34a084-fb80-4892-aa42-47860329e002">

### Fixed Issues
$ https://github.com/Expensify/App/issues/43199 
$ https://github.com/Expensify/App/issues/42082
$ https://github.com/Expensify/App/issues/43323

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
A lot of tests had to be adjusted because of the changes to the logic (only treating "&gt; " as a blockquotes and allowing blank lines in other cases)

/__tests__/ExpensiMark-HTML-test.js
updated:
- `Test quotes markdown replacement with quotes ends with blank quote rows`
- `Test quotes markdown replacement with quotes includes a middle blank quote row`
- `Test quotes markdown replacement with quotes includes multiple middle blank quote rows`
- `Test quotes markdown replacement with text includes blank quotes`
- `Test quotes markdown replacement with heading inside`
- `Test markdown quotes without spaces after > should not be parsed`
- `Test quotes markdown replacement with text includes multiple spaces`
- `test max level of blockquote (3)`

added:
- `Test quotes markdown with blank last line`
- `Test quotes markdown with last entry lacking space`

1. What tests did you perform that validates your changed worked?

- copy the code for ExpensiMark and replace it in the NewDot node modules 
- send a message presented below

```
> > > test
> > test2
> 
```

- verify that the empty blockquote in the last row is being displayed correctly

# QA
1. What does QA need to do to validate your changes?
Steps described in tests sections

1. What areas to they need to test for regressions?
Everything related to the empty blockquotes, but most importantly:

- mixing blockquotes with standard messages
- nesting blockquotes
